### PR TITLE
fix: where clause improvements

### LINKED ISSUE #9302: [Bug]:  S3 : First filter must be not be allowed to delete
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When the S3 fetch method is added it is observed that the first where clause can be deleted and the condition remains on undeleted. Making this UI difficult 

### Steps To Reproduce

1) Add Query S3 fetch method 
2) Add a Group filter 
3) Delete 1st where clause 
4) and observe 
Screenshot 2021-11-23 at 3 54 09 PM

### Environment

Release

### Version

Cloud

### PR DESCRIPTION
**Pull request by rashmi-rai**
New where clause improvements as per https://www.figma.com/file/Qs7p5frNx7uEfVYpADGk0d/Query-Pane-Improvements?node-id=323%3A17611

## Description

Fixes #9302 

## Type of change

- UI improvement

## How Has This Been Tested?

Tested Manually

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/where-clause 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.77 **(0)** | 37.16 **(-0.05)** | 35.35 **(-0.01)** | 56.02 **(0)**
 :green_circle: | app/client/src/components/formControls/SortingControl.tsx | 56.76 **(1.76)** | 56 **(-3.26)** | 11.11 **(0)** | 56.76 **(1.76)**
 :red_circle: | app/client/src/components/formControls/WhereClauseControl.tsx | 35.14 **(2.28)** | 22.54 **(-13.36)** | 0 **(0)** | 35.14 **(2.28)**
 :red_circle: | app/client/src/utils/WorkerUtil.ts | 88.98 **(-0.78)** | 70.59 **(-1.96)** | 100 **(0)** | 92.38 **(-0.95)**</details>

### CODE REVIEW

**Review by AmanAgarwal041 — APPROVED** [review]
LGTM

**Comment by ayushpahwa** [comment]
image
Should we need an outer box here?

**Comment by ayushpahwa** [comment]
image
This text seems to be wrong here

**Comment by ayushpahwa** [comment]
image
Just a question here. Should we allow for the first element to be deleted as well?

File: app/client/src/components/formControls/WhereClauseControl.tsx [hunk 1]
```diff
@@ -238,103 +258,116 @@ function ConditionBlock(props: any) {
...
         >
-          {/*Hardcoded label to be removed */}
-          <Text type={TextType.H5}>+ Add Filter</Text>
+          <Icon name="add-more-fill" size={IconSize.XL} />
+          <span style={{ marginLeft: "8px" }}>Add Where Condition</span>
         </AddMoreAction>
         {/* Check if the config allows more nesting, if it does, allow for adding more blocks */}
-        {props.currentNestingLevel < props.nestedLevels && (
+        <StyledTooltip
+          content="For S3 only 4 nested where condition group is allowed"
```
Comment (ayushpahwa) [inline app/client/src/components/formControls/WhereClauseControl.tsx:347]: This should be a dynamic text

**Review by ayushpahwa — CHANGES_REQUESTED** [review]

File: app/client/src/components/formControls/WhereClauseControl.tsx (hunk 1 above)
Comment (rashmi-rai) [reply app/client/src/components/formControls/WhereClauseControl.tsx:371]: We are not supporting using tooltips, labels, etc in composite components right now. This issue should be picked separately.

**Comment by rashmi-rai** [comment]
> image
> Should we need an outer box here?

Here we are expecting that user will add a group of conditions, hence the box.

**Comment by rashmi-rai** [comment]
> image
> This text seems to be wrong here

What am I missing here?

**Comment by rashmi-rai** [comment]
> image
> Just a question here. Should we allow for the first element to be deleted as well?

Look at the latest deploy preview and let me know if you still think so.

**Review by AmanAgarwal041 — APPROVED** [review]

**Comment by AmanAgarwal041** [comment]
There are some hard coded css which is present on the specific control components. Those will be solved in the issue #12159 with respect to the ads theming.

**Review by AmanAgarwal041 — APPROVED** [review]

**Pull request merged**